### PR TITLE
[feat] Add AimCallback for LightGBM

### PR DESCRIPTION
**Pull request by devfox-se**

### CODE REVIEW

**Comment by devfox-se** [comment]
Implements #1661

File: aim/sdk/adapters/lightgbm.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
+from typing import Optional
+
+from aim import Run
+from aim.ext.resource import DEFAULT_SYSTEM_TRACKING_INT
+
+try:
+    from lightgbm.callback import CallbackEnv
+except ImportError:
+    raise RuntimeError(
```
Comment (mahnerak) [inline aim/sdk/adapters/lightgbm.py:9]: This one will always warn (even if you don't use LightGBM adapter) doesn't it?

Comment (alberttorosyan) [reply aim/sdk/adapters/lightgbm.py:9]: @mahnerak error will be raised only if you try to import the adapter. as far as we don't have any imports in `aim` or `aim.sdk` modules, we should be fine.
@devfox-se please confirm

File: aim/sdk/adapters/lightgbm.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                 log_system_params: Optional[int] = True):
+        self._repo_path = repo
+        self._experiment = experiment
+        self._system_tracking_interval = system_tracking_interval
+        self._log_system_params = log_system_params
+        self._run = None
+        self._run_hash = None
+
+        # callback parameters
+        self.order = 15
```
Comment (alberttorosyan) [inline aim/sdk/adapters/lightgbm.py:35]: what `order` stands for?

File: aim/sdk/adapters/lightgbm.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+                data_name, eval_name, result, _ = item
+                self._run.track(result, name=eval_name, context={'data_name': data_name})
+            else:
+                data_name, eval_name = item[1].split()
+                res_mean = item[2]
+                res_stdv = item[4]
+                self._run.track(res_mean, name=f'{eval_name}-mean', context={'data_name': data_name})
+                self._run.track(res_stdv, name=f'{eval_name}-stdv', context={'data_name': data_name})
+
+        self.before_tracking(env=env)
```
Comment (alberttorosyan) [inline aim/sdk/adapters/lightgbm.py:88]: should be `after_tracking` I guess

File: docs/source/using/integration_guides.md [hunk 4]
```diff
@@ -184,3 +184,22 @@ class CustomCallback(AimCallback):
...
+
+class CustomCallback(AimCallback):
+
+    def before_tracking(self, env):
+        for item in env.evaluation_result_list:
+            # manipulate item here
+            pass
+
+    def after_tracking(self, env):
+        self.experiment.track(env.iteration, name="Iteration")
```
Comment (alberttorosyan) [inline docs/source/using/integration_guides.md:204]: is this the iteration step? what about adding some actual metric or maybe other objects?

**Review by alberttorosyan — CHANGES_REQUESTED** [review]
left a couple of comments; overall looks good!
I think we can also add the LigtGBM logo in docs landing page and GH readme 🤔

File: aim/sdk/adapters/lightgbm.py (hunk 3 above)
Comment (devfox-se) [reply aim/sdk/adapters/lightgbm.py:88]: good catch

File: aim/sdk/adapters/lightgbm.py (hunk 1 above)
Comment (devfox-se) [reply aim/sdk/adapters/lightgbm.py:9]: no there is no problem with it, we have this technique used in all of our callbacks

File: aim/sdk/adapters/lightgbm.py (hunk 2 above)
Comment (devfox-se) [reply aim/sdk/adapters/lightgbm.py:35]: I cannot really say, its not documented and all the `LightGBM`'s callbacks have order set in them, I picked a value in between

**Comment by devfox-se** [comment]
@alberttorosyan Regarding the landing page logo - we need something prepared for it to align with others

File: docs/source/using/integration_guides.md (hunk 4 above)
Comment (devfox-se) [reply docs/source/using/integration_guides.md:204]: yeah its really hard to come up with something more useful 😆

File: aim/sdk/adapters/lightgbm.py (hunk 2 above)
Comment (alberttorosyan) [reply aim/sdk/adapters/lightgbm.py:35]: @mahnerak any idea here?

Comment (mahnerak) [reply aim/sdk/adapters/lightgbm.py:35]: When there are multiple Callbacks registered this defines the precedence of their calls.
I think this value should be as great (as late) as possible, but before checkpointing.

It looks like `30` is for early stopping and serialization of the model, and `20` for recording some logs. So something in between will work better.

Comment (alberttorosyan) [reply aim/sdk/adapters/lightgbm.py:35]: @devfox-se please double-check this.

**Review by alberttorosyan — APPROVED** [review]

**Pull request merged**